### PR TITLE
Compaction Scheduling

### LINKED ISSUE #1718: FR: compacting schedule like we do for pruning
#### The problem

I just recently upgraded to borg 1.2. The repo started growing very fast, and I learn that now we need to run compact at regular intervals to reclaim space. The compact button allowed me to recover the lost space for my repo. But I feel it should be run regularly, like we do for pruning or validation

#### Requested Solution

Having the possibility to compact after each backup or at regular interval.

#### Additional context

I saw some discussions about this in other issues about compact, but nothing clearly about this. Maybe I missed something?

### PR DESCRIPTION
**Pull request by shivansh02**
### Description

Allows users to run `borg compact` after specified intervals. Added 2 columns to the profile model: `compaction_on` to check/uncheck scheduling, and `compaction_weeks` to set interval in weeks. Default interval set to 3 weeks.

### Related Issue

Resolves #1718 
### Motivation and Context

Backup space usage is exploding for some people until they realize they have to run compact manually. This allows users to schedule running compact just like validation.

### How Has This Been Tested?

Tested manually by setting `compaction_cutoff`  to minutes instead of days in `scheduler.py`.
### Screenshots (if appropriate):
Screenshot from 2024-04-05 14-04-56
Screenshot from 2024-04-05 14-09-33

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] I have read the [CONTRIBUTING](https://vorta.borgbase.com/contributing/) guide.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

*I provide my contribution under the terms of the [license](./../LICENSE.txt) of this repository and I affirm the [Developer Certificate of Origin][dco].*

[dco]: https://developercertificate.org/

### CODE REVIEW

File: src/vorta/store/migrations.py [hunk 1]
```diff
@@ -248,6 +248,16 @@ def run_migrations(current_schema, db_connection):
...
+            migrator.add_column(
+                BackupProfileModel._meta.table_name,
+                'compaction_on',
+                pw.BooleanField(default=True),
+            ),
+            migrator.add_column(
+                BackupProfileModel._meta.table_name,
+                'compaction_weeks',
+                pw.IntegerField(default=3),
+            ),
```
Comment (real-yfprojects) [inline src/vorta/store/migrations.py:260]: Afaik a new scheme version should be introduced for that.

File: src/vorta/scheduler.py [hunk 2]
```diff
@@ -487,6 +489,24 @@ def post_backup_tasks(self, profile_id):
             if msg['ok']:
                 job = BorgCheckJob(msg['cmd'], msg, profile.repo.id)
                 self.app.jobs_manager.add_job(job)
+        
+        # Check if a compact job is needed
```
Comment (real-yfprojects) [inline src/vorta/scheduler.py:493]: Please update doc string of this method.

**Review by real-yfprojects — COMMENTED** [review]
Thanks for this nice addition!

**Comment by shivansh02** [comment]
@m3nu All checks passed, can we merge this now?

File: src/vorta/store/models.py [hunk 3]
```diff
@@ -90,6 +90,8 @@ class BackupProfileModel(BaseModel):
     schedule_make_up_missed = pw.BooleanField(default=True)
     validation_on = pw.BooleanField(default=True)
     validation_weeks = pw.IntegerField(default=3)
+    compaction_on = pw.BooleanField(default=True)
```
Comment (m3nu) [inline src/vorta/store/models.py:93]: Let's disable it by default to avoid surprising behavior.

**Review by m3nu — CHANGES_REQUESTED** [review]

File: src/vorta/store/models.py (hunk 3 above)
Comment (shivansh02) [reply src/vorta/store/models.py:93]: Sure!

**Review by m3nu — APPROVED** [review]

**Pull request merged**